src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.cs

```diff
@@ -263,18 +263,21 @@ public override void VisitNamedType(NamedTypeSymbol symbol)
             void appendContainedExtensions(NamedTypeSymbol containingType)
             {
                 Debug.Assert(!_isForSingleSymbol);
-                ExtensionGroupingInfo extensionGroupingInfo = ((SourceMemberContainerTypeSymbol)containingType).GetExtensionGroupingInfo();
-
-                foreach (ImmutableArray<SourceNamedTypeSymbol> extensions in extensionGroupingInfo.EnumerateMergedExtensionBlocks())
+                if (containingType is SourceMemberContainerTypeSymbol sourceContainingType)
```

Comment:
>if (containingType is SourceMemberContainerTypeSymbol sourceContainingType) [](http://example.com/codeflow?start=16&length=75)

This doesn't look like a proper fix. This just ignores extensions from added modules, however it looks like declarations in added modules aren't ignored in general.

Reply:
My understanding is that added modules cannot contribute any doc comments. It was not my goal to address this bigger question in this PR, but adding `if (symbol is not SourceNamedTypeSymbol) { return; }` in `VisitNamedType` indeed didn't break any test.

Reply:
> My understanding is that added modules cannot contribute any doc comments.

If that is what we think, then a proper fix, in my opinion, would be to stop visiting declarations from added modules in this component. Where exactly the change should be made can be discussed off-line. I do not think it should be the change that was tried. Quite possibly it belongs in the caller of this component.

However, if we think that added modules can contribute doc comments , then most likely extensions should contribute as well, and the change in commit 1 is not correct. More specifically, it does prevent the crash, but most likely doesn't result in correct behavior.